Add missing fuzzers from cncf-fuzzing project

joestringer opened the pull request:
Originally authored by @AdamKorcz, pulled from github.com/cncf/cncf-fuzzing .

This PR also converts all fuzzers to native Go fuzzers, making it easier to
reproduce fuzz results using go tooling (ie `go test -fuzz ...`).

joestringer committed:
fuzz: Convert remaining fuzzers to native Go

Convert the fuzzing signatures to native Go fuzzer signatures, and
switch the invocation in the oss-fuzz wrapper script to use the native
Go fuzzer invoker. While we're at it, fix up any remaining quirks like
additional parameters that should also be fuzzed.

Signed-off-by: Joe Stringer <joe@cilium.io>

joestringer committed:
Add missing fuzzers from cncf-fuzzing project

Originally authored by AdamKorcz, pulled from github.com/cncf/cncf-fuzzing

Modified from original submissions to compile against the latest code.

Co-authored-by: AdamKorcz <adam@adalogics.com>
Signed-off-by: AdamKorcz <adam@adalogics.com>
Signed-off-by: Joe Stringer <joe@cilium.io>

bimmlerd commented on pkg/labelsfilter/fuzz_test.go:
```diff
@@ -7,60 +7,67 @@ import (
-	if err != nil {
-		return 0
-	}
+		file, err := os.Create("file")
```
Does this leave files lying around? Could either employ `t.Cleanup()` to clean it up or place the file in https://pkg.go.dev/testing#T.TempDir so that it is cleaned up automatically

joestringer replied:
Good suggestion :+1:

joestringer replied:
Applied a new commit on top:

```
diff --git a/pkg/labelsfilter/fuzz_test.go b/pkg/labelsfilter/fuzz_test.go
index 195b5c4dcd46..149556c144c2 100644
--- a/pkg/labelsfilter/fuzz_test.go
+++ b/pkg/labelsfilter/fuzz_test.go
@@ -7,6 +7,7 @@ import (
        "fmt"
        "log/slog"
        "os"
+       "path/filepath"
        "testing"
 
        fuzz "github.com/AdaLogics/go-fuzz-headers"
@@ -52,7 +53,9 @@ func FuzzLabelsfilterPkg(f *testing.F) {
 
                lbls := labels.Map2Labels(stringMap, source)
 
-               file, err := os.Create("file")
+               baseDir := t.TempDir()
+               path := filepath.Join(baseDir, "cilium_fuzz_labelsfilter")
+               file, err := os.Create(path)
                if err != nil {
                        return
                }
@@ -63,7 +66,7 @@ func FuzzLabelsfilterPkg(f *testing.F) {
                        return
                }
 
-               err = ParseLabelPrefixCfg(slog.New(slog.DiscardHandler), prefixes, nodePrefixes, "file")
+               err = ParseLabelPrefixCfg(slog.New(slog.DiscardHandler), prefixes, nodePrefixes, path)
                if err != nil {
                        fmt.Println(err)
                        return
```

bimmlerd reviewed (approved):
LGTM. Could clean up files which the fuzzer seems to leave behind, but AFAIK fuzzers don't get run by default, so it's not a big concern.

joestringer committed:
labelsfilter: Clean up fuzzer file after test

This test was not being cleaned up after running the fuzzer. Use
testing.T.TempDir() to ensure it is cleaned up.

Suggested-by: David Bimmler <david.bimmler@isovalent.com>
Signed-off-by: Joe Stringer <joe@cilium.io>
